projects/cds/src/profiletag/services/profiletag-events.ts

```diff
@@ -0,0 +1,120 @@
+import { isPlatformBrowser } from '@angular/common';
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { BaseSiteService, WindowRef } from '@spartacus/core';
+import { fromEvent, merge, Observable } from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  map,
+  switchMap,
+  take,
+  tap,
+} from 'rxjs/operators';
+import { CdsConfig } from '../../config/index';
+import {
+  ConsentReferenceEvent,
+  DebugEvent,
+  ProfileTagEventNames,
+  ProfileTagJsConfig,
+  ProfileTagWindowObject,
+} from '../model/profile-tag.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProfileTagEventTracker {
```

Comment:
Please remove `public` keyword, as in TS everything that doesn't have a _modifier_ is `public` by default.

Reply:
Also, please make methods `private` that are not needed to be called from outside.
If testing/calling private methods is an issue, then you can use this trick: https://stackoverflow.com/a/53483320/5252849. I suggest to create constants that will hold the method names, and then reference those; in case the method name changes in the future, it's easy to go back and update the value of one constant.